catkit2/services/oceanoptics_spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,187 @@
+        A NDArray with the wl of the spectrometer.
+    spectras : DataStream
+        A data stream to submit the spectras from the spectrometer.
+    should_be_acquiring : threading.Event
+        An event to signal whether the spectra should acquire a spectrum.
+    NUM_FRAMES : int
+        The number of frames to allocate for the data streams.
+
+    Methods
+    -------
+    open()
+        Open the service.
+    main()
+        The main function of the service.
+    close()
+        Close the service.
+    start_acquisition()
+        Start the acquisition.
+    take_one_spectrum()
+        Get a spectrum from the spectrometer.
+    exposure_time()
+        Set the exposure time of the spectrometer.
+
+    '''
+    NUM_FRAMES = 20
+
+    def __init__(self):
+        '''
+        Create a new OceanOpticsSpectrometer service.
+
+        '''
+        super().__init__('oceanoptics_spectrometer')
+        
+        self.serial_number = None
+        self.spectrometer = None
+        self.model = None
+        self.pixels_number = None
+
+        self.wavelengths = None
+        self.spectras = None
+
+        self._exposure_time = None
+
+        self.should_be_acquiring = threading.Event()
+        self.should_be_acquiring.set()
+
+    def open(self):
+        '''
+        Open the service.
+
+        This function is called when the service is opened.
+        It initializes the spectrometer and creates the data streams and properties.
+
+        Raises
+        ------
+        RuntimeError
+            If the spectro cannot be found.
+
+        '''
+
+        # Find the spectrometer
+        self.serial_number = str(self.config['serial_number'])
+        try:
+            self.spectrometer = spct.from_serial_number(self.serial_number)
+        except SeaBreezeError:
+            raise ImportError(f'OceanOptics: Could not find spectrometer with serial number {self.serial_number}')
+
+        # exctract attributes
+        self.model = self.spectrometer.model
+        self.pixels_number = self.spectrometer.pixels
+        self.wavelengths = self.spectrometer.wavelengths()
+
+        # Define and set defaut exposure time
+        self._exposure_time = self.config.get('exposure_time', 1000)
+        self.exposure_time(self._exposure_time)
+
+        # Create datastreams
+        # Use the full sensor size here to always allocate enough shared memory.
+        self.spectras = self.make_data_stream('spectras', 'float32', [self.pixels_number], self.NUM_FRAMES)
+
+    def main(self):
+        '''
+        The main function of the service.
+
+        This function is called when the service is started.
+        '''
+        while not self.should_shut_down:
+            if self.should_be_acquiring.wait(0.05):
+                self.take_one_spectrum()
+                self.should_be_acquiring.clear()
+
+    def start_acquisition(self):
+        '''
+        Start the acquisition loop.
+        '''
+        self.should_be_acquiring.set()
```

Comment:
If you have a start_acquisition() method, you should also have an end_acquisition() method like here:
https://github.com/spacetelescope/catkit2/blob/7663307222cee45655932ded1887ac10e8065d90/catkit2/services/allied_vision_camera/allied_vision_camera.py#L319

and use that in your while loop in main(). *If* you want to have a continuous acquisition mode.